File: userdataapp-frontend-vite/src/types/userData.ts
export interface UserData {
  firstName: string;
  lastName: string;
  email: string;
  title: string;
  gender: string;
  country: string;
  registrationDate: string;
  birthDate: string;
  salary: number;
  comments: string;
}

export interface PaginatedResponse<T> {
  data: T[];
  totalCount: number;
  totalPages: number;
  currentPage: number;
  pageSize: number;
}

export interface FilterOptions {
  gender?: string;
  country?: string;
  registrationDateFrom?: string;
  registrationDateTo?: string;
  birthDateFrom?: string;
  birthDateTo?: string;
  minSalary?: number;
  maxSalary?: number;
}

export interface SearchOptions {
  firstName?: string;
  lastName?: string;
  email?: string;
  title?: string;
  comments?: string;
}

File: UserDataApp.API/Models/UserData.cs
using System;

namespace UserDataApp.API.Models
{
  public class UserData
  {
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Gender { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Comments { get; set; } = string.Empty;
    public DateTime RegistrationDate { get; set; }
    public DateTime BirthDate { get; set; }
    public decimal Salary { get; set; }
  }
}